### ISSUE TITLE
[iOS] “Remote service error” while trying to log into VPN with an account that had a subscription made via IAP

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/mozilla-vpn-client#1771]
**VPN version:**

* 2.5.0 (2.202109031936);

**Affected Platforms:**

* iOS;

**Prerequisites:**

* Have one FxA account with IAP subscription that expired for Mozilla VPN;
* Have the VPN installed on your machine;

**Steps to reproduce:**

1. Open VPN
1. Login to VPN using the account from prerequisites;
1. Observe the behavior;

**Expected result:**
User is redirected to the subscription page or a suggestive message is displayed.

**Actual result:**
“Remote service error” red banner is displayed after sign in.

**Notes:**

* 【 highlighted #ff5630 】This is major issue because the user that subscribes to VPN via IAP might not be able to use it on other platforms. We are not sure if this is an only-stage issue caused by specific stage configurations.【 end highlighted 】
* not reproducing is the subscription is made on Android IAP;
* Attaching logs from Windows 10. (I also reproduced a crash on Win while doing this - but I do not have steps)

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/VPN-973)
┆Reporter: Valentina Virlics
┆Link To Issue: https://mozilla-hub.atlassian.net/browse/VPN-973

**Comment by mbirghan** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Look into iOS IAP issue VPN-973](https://trello.com/c/l51S0iNa/44-look-into-ios-iap-issue-vpn-973)

**Comment by data-sync-user** [comment]
➤ Moritz Birghan commented:

Valentina Virlics do you also have logs for iOS?

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Moritz Birghan https://controlc.com/0b440cd3 ( https://controlc.com/0b440cd3|smart-link )  here are the iOS logs.

**Comment by data-sync-user** [comment]
➤ Sarah Bird commented:

Valentina Virlics I am not able to reproduce the bug. These are the steps I took, all on iOS:

  * Got app from Testflight
  * Switched to staging server mode
  * Signed in with new fxa account
  * Was prompted for subscription
  * Selected monthly subscription
  * Subscribed
  * I couldn’t find a way to cancel the subscription so in the end I just waited and it seemed to expire by itself.
  * I uninstalled the app and reinstalled the testflight app
  * Switched to staging server mode
  * Signed in with the same fxa account
  * I was prompted for a subscription

I am on very latest testflight build that was pushed just recently - 2.202109091821

Can you please re-verify that this is a reproducible bug for you? … Or tell me what step I’m doing wrong to reproduce.

**Comment by data-sync-user** [comment]
➤ Pun Lam commented:

!IMG_06479F7AB0A9-1.jpeg|width=375,height=812!

**Comment by data-sync-user** [comment]
➤ Pun Lam commented:

Not sure it is the same problem with the Aug 6 review.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Pun Lam Sarah Bird unfortunately we were blocked on checking this by https://mozilla-hub.atlassian.net/browse/VPN-982 ( https://mozilla-hub.atlassian.net/browse/VPN-982|smart-link ) .

What I can say, is that, on the previous build, I waited around 1 hour to expire the subscription, and tried after.

**Comment by data-sync-user** [comment]
➤ Sarah Bird commented:

Thanks for the update. Sorry I broke everything. It sounds like we did the same thing but I couldn’t repro. Let me know on Monday if you can repro.

**Comment by data-sync-user** [comment]
➤ Sarah Bird commented:

Adrienne Davenport Pun Lam Andrea Marchesini - if valentina dose repro this bug on monday we should have a conversation about it’s priority for v2.5.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Verified this as fixed on the last iOS build - 2.202109131634. 

After the subscription expires, user can subscribe again using the same account. Tried on different platforms.

**Comment by data-sync-user** [comment]
➤ Sarah Bird commented:

Just to be clear. No fix was provided for this. It was a transient error that I could never reproduce.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

The initial verification of the issue, did not reveal the error. But, after further investigation, we were able to trigger the error message. 
If it is a wrong behavior and you think a separate issue should be logged, please let me know.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

For some reason, only a part of my previous comment was recorded.

!image.png|width=1143,height=438!

**Comment by data-sync-user** [comment]
➤ Sarah Bird commented:

Valentina Virlics please check if this is also new on v2.4.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Verified this on VPN 2.4 using a Stage build. The error is also displayed.

The interesting thing is that doing the sign in with the second account (the one without subscription) on a 2.5 build on Windows, I can see the same error, as long as the subscription for the first account is still active on iOS. Sorry if I am not very clear in explanations.

**Comment by birdsarah** [comment]
@punster-moz - should this issue be closed on gh?

**Comment by data-sync-user** [comment]
➤ Pun Lam commented:

What is the conclusion ? Is it caused by multiple login-in accounts which will not happen for a regular user ?

I wonder if setting the status to done, will close it on gh.

**Comment by data-sync-user** [comment]
➤ Sarah Bird commented:

The conclusions are that:

  * this is a real issue
  * it has been present since at least 2.4 so is not a 2.5 / Android IAP introduced bug so it’s probably not that important. 
  * something weird is going on with Jira sync as i don’t think this should be closed on gh.

**Comment by data-sync-user** [comment]
➤ Sarah Bird commented:

Adrienne Davenport can you re-confirm the categorization of this issue:

  * 2.5 is done
  * is it really “high” priority when it’s been an issue since at least 2.4 and was only detected in QA for in app purchase

I’m going to un-assign myself as there’s nothing about the in app purchase code I wrote that caused it, so anyone can pick it up.

**Comment by data-sync-user** [comment]
➤ Adrienne Davenport commented:

Priority updated to Low as this is not a prod issue.

**Comment by data-sync-user** [comment]
➤ Sarah Bird commented:

Valentina Virlics can we close this issue?
